packages/shared_preferences/shared_preferences_linux/lib/shared_preferences_linux.dart

```diff
@@ -148,13 +117,296 @@ class SharedPreferencesLinux extends SharedPreferencesStorePlatform {
+    String key,
+    String value,
+    SharedPreferencesOptions options,
+  ) =>
+      _setValue(key, value, options);
+
+  @override
+  Future<void> setBool(
+    String key,
+    bool value,
+    SharedPreferencesOptions options,
+  ) =>
+      _setValue(key, value, options);
+
+  @override
+  Future<void> setDouble(
+    String key,
+    double value,
+    SharedPreferencesOptions options,
+  ) =>
+      _setValue(key, value, options);
+
+  @override
+  Future<void> setInt(
+    String key,
+    int value,
+    SharedPreferencesOptions options,
+  ) =>
+      _setValue(key, value, options);
+
+  @override
+  Future<void> setStringList(
+    String key,
+    List<String> value,
+    SharedPreferencesOptions options,
+  ) =>
+      _setValue(key, value, options);
+
+  @override
+  Future<String?> getString(
+    String key,
+    SharedPreferencesOptions options,
+  ) async {
+    final Map<String, Object> data =
+        await _readAllFromFile(<String>{key}, options);
+    return data[key] as String?;
+  }
+
+  @override
+  Future<bool?> getBool(
+    String key,
+    SharedPreferencesOptions options,
+  ) async {
+    final Map<String, Object> data =
+        await _readAllFromFile(<String>{key}, options);
+    return data[key] as bool?;
+  }
+
+  @override
+  Future<double?> getDouble(
+    String key,
+    SharedPreferencesOptions options,
+  ) async {
+    final Map<String, Object> data =
+        await _readAllFromFile(<String>{key}, options);
+    return data[key] as double?;
+  }
+
+  @override
+  Future<int?> getInt(
+    String key,
+    SharedPreferencesOptions options,
+  ) async {
+    final Map<String, Object> data =
+        await _readAllFromFile(<String>{key}, options);
+    return data[key] as int?;
+  }
+
+  @override
+  Future<List<String>?> getStringList(
+    String key,
+    SharedPreferencesOptions options,
+  ) async {
+    final Map<String, Object> data =
+        await _readAllFromFile(<String>{key}, options);
+    return data[key] as List<String>?;
+  }
+
+  @override
+  Future<void> clear(ClearPreferencesParameters parameters,
+      SharedPreferencesOptions options) async {
+    final SharedPreferencesLinuxOptions linuxOptions =
+        SharedPreferencesLinuxOptions.fromSharedPreferencesOptions(options);
+    final PreferencesFilters filter = parameters.filter;
+    final Map<String, Object> preferences =
+        await _readPreferences(linuxOptions.fileName);
+    preferences.removeWhere((String key, _) =>
+        filter.allowList == null || filter.allowList!.contains(key));
+    await _writePreferences(
+      preferences,
+      linuxOptions.fileName,
+      fs: fs,
+      pathProvider: pathProvider,
+    );
+  }
+
+  @override
+  Future<Map<String, Object>> getPreferences(
+    GetPreferencesParameters parameters,
+    SharedPreferencesOptions options,
+  ) async {
+    return _readAllFromFile(parameters.filter.allowList, options);
+  }
+
+  Future<Map<String, Object>> _readAllFromFile(
```

Comment:
We should rename this; I thought it was reading directly from the file, but it actually uses the cache if the cache has been made.